int-set/src/bitset.rs

```diff
@@ -252,6 +463,23 @@ impl BitSet {
         let page_index = self.ensure_page_index_for_major(major_value);
         self.pages.get_mut(page_index).unwrap()
     }
+
+    // Return the mutable page at a given index
+    fn page_for_index_mut(&mut self, index: usize) -> Option<&mut BitPage> {
+        if let Some(page_info) = self.page_map.get(index) {
+            self.pages.get_mut(page_info.index as usize)
+        } else {
+            None
+        }
```

Comment:
could just write this as,

```suggestion
        self.page_map
            .get(index)
            .and_then(|info| self.pages.get_mut(info.index as usize))
```